scripts/download_certificate.py

```diff
@@ -4,23 +4,31 @@
 
 from OpenSSL import crypto
 import os
+import time
 
 hostname = os.uname()[1]
 
 cfg = Config()
 openssl = OpenSSLWrapper()
 
-kvwrapper = KeyVaultWrapper(
-    client_id = cfg.client_id, 
-    secret = cfg.secret,
-    tenant_id = cfg.tenant_id)
+def load_data():
+    kvwrapper = KeyVaultWrapper()
 
-kvwrapper.get_secret_to_file(cfg.vault_uri, 'root-ca', '', 'root_ca.pem')
-kvwrapper.get_secret_to_file(cfg.vault_uri, 'intermediate-ca', '', 'intermediate_ca.pem')
+    kvwrapper.get_secret_to_file(cfg.vault_uri, 'root-ca', '', 'root_ca.pem')
+    kvwrapper.get_secret_to_file(cfg.vault_uri, 'intermediate-ca', '', 'intermediate_ca.pem')
 
-kvwrapper.get_certificate_and_key_to_file(
-    vault_uri = cfg.vault_uri,
-    cert_name = hostname,
-    cert_version = '',
-    key_filename = 'new.key',
-    cert_filename = 'new.pem')
+    kvwrapper.get_certificate_and_key_to_file(
+        vault_uri = cfg.vault_uri,
+        cert_name = hostname,
+        cert_version = '',
+        key_filename = 'new.key',
+        cert_filename = 'new.pem')
+
+while True:

```

Comment:
Possible infinite loop if connection to KV is down?

Reply:
Yes, but if KV is down your VM is useless.